rclc_parameter/src/rclc_parameter/parameter_server.c

```diff
@@ -0,0 +1,758 @@
+
+  ret &= rcl_service_fini(&parameter_server->list_service, node);
+  ret &= rcl_service_fini(&parameter_server->set_service, node);
+  ret &= rcl_service_fini(&parameter_server->get_service, node);
+  ret &= rcl_service_fini(&parameter_server->get_types_service, node);
+  ret &= rcl_service_fini(&parameter_server->describe_service, node);
+
+  if (parameter_server->notify_changed_over_dds) {
+    ret &= rcl_publisher_fini(&parameter_server->event_publisher, node);
+  }
+
+  return ret;
+}
+
+rcl_ret_t
+rclc_executor_add_parameter_server(
+  rclc_executor_t * executor,
+  rclc_parameter_server_t * parameter_server,
+  ModifiedParameter_Callback on_modification)
+{
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    executor, "executor is a null pointer", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    parameter_server, "parameter_server is a null pointer", return RCL_RET_INVALID_ARGUMENT);
+
+  rcl_ret_t ret;
+
+  parameter_server->on_modification = on_modification;
+
+  ret = rclc_executor_add_service_with_context(
+    executor, &parameter_server->list_service,
+    &parameter_server->list_request, &parameter_server->list_response,
+    rclc_parameter_server_list_service_callback, parameter_server);
+
+  ret &= rclc_executor_add_service_with_context(
+    executor, &parameter_server->get_types_service,
+    &parameter_server->get_types_request, &parameter_server->get_types_response,
+    rclc_parameter_server_get_types_service_callback, parameter_server);
+
+  ret &= rclc_executor_add_service_with_context(
+    executor, &parameter_server->set_service,
+    &parameter_server->set_request, &parameter_server->set_response,
+    rclc_parameter_server_set_service_callback,
+    parameter_server);
+
+  ret &= rclc_executor_add_service_with_context(
+    executor, &parameter_server->get_service,
+    &parameter_server->get_request, &parameter_server->get_response,
+    rclc_parameter_server_get_service_callback,
+    parameter_server);
+
+  ret &= rclc_executor_add_service_with_context(
+    executor, &parameter_server->describe_service,
+    &parameter_server->describe_request, &parameter_server->describe_response,
+    rclc_parameter_server_describe_service_callback,
+    parameter_server);
+
+  return ret;
+}
+
+rcl_ret_t
+rclc_add_parameter(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name,
+  rclc_parameter_type_t type)
+{
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    parameter_server, "parameter_server is a null pointer", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    parameter_name, "parameter_name is a null pointer", return RCL_RET_INVALID_ARGUMENT);
+
+  size_t index = parameter_server->parameter_list.size;
+
+  if (index >= parameter_server->parameter_list.capacity ||
+    rclc_parameter_search(&parameter_server->parameter_list, parameter_name) != NULL)
+  {
+    return RCL_RET_ERROR;
+  }
+  rclc_parameter_set_string(&parameter_server->parameter_list.data[index].name, parameter_name);
```

Comment:
```suggestion
 if (! rclc_parameter_set_string(&parameter_server->parameter_list.data[index].name, parameter_name))
{
  return RCL_RET_ERROR;
}
```
Check for parameter_name size